app/src/main/java/com/hedvig/app/feature/crossselling/ui/CrossSellingResultScreen.kt

```diff
@@ -0,0 +1,180 @@
+import androidx.compose.ui.tooling.preview.PreviewParameter
+import androidx.compose.ui.tooling.preview.PreviewParameterProvider
+import androidx.compose.ui.unit.dp
+import com.hedvig.app.R
+import com.hedvig.app.ui.compose.HedvigTheme
+import com.hedvig.app.ui.compose.designsystem.LargeContainedButton
+import com.hedvig.app.ui.compose.designsystem.LargeOutlinedButton
+import java.time.LocalDate
+import java.time.format.DateTimeFormatter
+
+// TODO text resources still missing
+@Composable
+fun CrossSellingResultScreen(
+    crossSellingResult: CrossSellingResult
+) {
+    HedvigTheme {
+        Box(
+            modifier = Modifier
+                .fillMaxSize()
+                .padding(horizontal = 16.dp)
+        ) {
+            InformationSection(
+                crossSellingResult = crossSellingResult,
+                modifier = Modifier
+                    .align(Alignment.TopStart)
+                    .padding(top = 96.dp)
+                    .fillMaxWidth(),
+            )
+            ButtonsSection(
+                crossSellingResult = crossSellingResult,
+                modifier = Modifier
+                    .align(Alignment.BottomStart)
+                    .padding(bottom = 16.dp)
+                    .fillMaxWidth(),
+            )
+        }
+    }
+}
+
+@Composable
+private fun InformationSection(
+    crossSellingResult: CrossSellingResult,
+    modifier: Modifier
+) {
+    @DrawableRes
+    val drawableId: Int = when (crossSellingResult) {
+        is CrossSellingResult.Success -> R.drawable.ic_checkmark_in_circle
+        CrossSellingResult.Error -> R.drawable.ic_x_in_circle
+    }
+    val titleText = when (crossSellingResult) {
+        CrossSellingResult.Error -> "Something went wrong."
+        is CrossSellingResult.Success -> {
+            stringResource(
+                R.string.purchase_confirmation_new_insurance_today_app_state_title,
+                crossSellingResult.insuranceType
+            )
+        }
+    }
+    val descriptionText = when (crossSellingResult) {
+        CrossSellingResult.Error -> "Your purchase couldn't be completed.\nContact us in the chat."
+        is CrossSellingResult.Success -> {
+            when {
+                crossSellingResult.startingDate.dayOfMonth <= LocalDate.now().dayOfMonth -> {
+                    stringResource(R.string.purchase_confirmation_new_insurance_today_app_state_description)
+                }
+                else -> {
+                    val activationDate = crossSellingResult.startingDate.format(
+                        DateTimeFormatter.ISO_LOCAL_DATE
+                    )
+                    stringResource(
+                        R.string.purchase_confirmation_new_insurance_active_in_future_app_state_description,
+                        activationDate
+                    )
+                }
+            }
+        }
+    }
+    Column(
+        modifier = modifier,
+        verticalArrangement = Arrangement.spacedBy(18.dp),
```

Comment:
Perhaps moreso a comment to the designers, but I'm not sure that `18 dp` is a size we want to use in practice. There are some issues with translating the spacing in Figma to implementation that I'd like to show you as well, in particular with regards to the concept of `line height`, which does not work as Figma thinks it does in android 🙄

Reply:
Yeah, I can show you how I got that number from Figma and you can tell me if I made some wrong assumptions on that process.